docs/data_model.md

```diff
@@ -0,0 +1,105 @@
+# Data model
+
+The Geneve data model describes what data Geneve is expected to generate,
+it guides and constraints the data generation process so that the output
+satisfies your criteria.
+
+Think in this way: data generation is a random process, at its root it
+just produces a long random string made of 0s and 1s. What you actually
+want is to shape the result and channel the randomness so that the
+generated data looks sensible in your context and at the same time never
+quite the same.
```

Comment:
Love this -- appreciate the first principals level-set here 🙂. The output (data generated) just needs to meet the minimum interface contract defined by the input query, and can _and should_ vary indiscriminately outside of that.

As a further abstraction (which could be an input to Geneve), have you had any thoughts around some sort of input definition spec or anything? This is probably most applicable to the "click through" example mentioned above, but could be useful when needing to generate multiple documents and managing the references and chained dependencies between each (e.g. capturing parent/child relationships, generating sub-documents that are in the correct subnet, etc). Supporting sequences in EQL is a sorta similar situation, albeit still defined and verifiable by the query spec.

Reply:
Ahhh, reading further looks like the `Data Model` defined below pretty much handles this. 👍 Would be nice to package these up ala [Rally Tracks](https://esrally.readthedocs.io/en/stable/adding_tracks.html) so it's easy to run pre-defined configurations.